File: 백준/Silver/8979. 올림픽/올림픽.py
import sys
input = sys.stdin.readline
n, k = map(int, input().split())
data = []
medal = []
for i in range(n):
    nation, gold, silver, copper = map(int, input().split())
    data.append((nation, gold, silver, copper))
    medal.append((gold * 100000000) + (silver * 10000) + (copper * 1))
medal = sorted(medal, reverse=True)
temp, ans = 0, 0
for j in data:
    if j[0] == k:
        temp = medal.index(((j[1] * 100000000) + (j[2] * 10000) + (j[3] * 1)))
        for k in range(temp, -1, -1):
            if medal[k] > ((j[1] * 100000000) + (j[2] * 10000) + (j[3] * 1)):
                ans += 1
        break
print(ans+1)

File: 백준/Silver/8979. 올림픽/올림픽.c
#include <stdio.h>

typedef struct {
    int num;
    int gold;
    int silver;
    int bronze;
} MEDAL;

MEDAL medal[1001];

int main(void) {
    
    int n, k;
    scanf("%d %d", &n, &k);       // 국가의 수, 등수를 알고 싶은 국가

    int target = 0;     // target index
    for (int i = 0; i < n; i++) {

        scanf("%d %d %d %d", &medal[i].num, &medal[i].gold, &medal[i].silver, &medal[i].bronze);

        if (medal[i].num == k) {
            target = i;
        }
    }

    int rank = 0;

    for (int i = 0; i < n; i++) {

        if (i != target) {
            if (medal[i].gold > medal[target].gold) {
                rank++;
            }
            else if (medal[i].gold == medal[target].gold) {
                if (medal[i].silver > medal[target].silver) {
                    rank++;
                }
                else if (medal[i].silver == medal[target].silver) {
                    if (medal[i].bronze > medal[target].bronze) {
                        rank++;
                    }
                }
            }
        }

    }

    printf("%d\n", rank + 1);

    return 0;
}